### PR TITLE
Enforce server version when using GSI

### LINKED ISSUE #3419: GSI: Add Vulcan version check
Add validation check to require Couchbase Server 5.5 or higher when using GSI.

Can leverage existing CouchbaseServerVersion functionality, but should consider:
- helper function to more easily compare major/minor versions
- specify a minimum version on the bucket spec, instead of useXattrs/useViews/etc

### PR DESCRIPTION
**Pull request by adamcfraser**
Fixes #3419

### CODE REVIEW

**Review by tleyden — APPROVED** [review]
LGTM

File: rest/server_context.go [hunk 1]
```diff
@@ -469,12 +469,23 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(config *DbConfig, useExisti
...
 	if !useViews {
+
+		// Couchbase Server version must be 5.5 or higher to use GSI
+		gsiSupported, errServerVersion := base.IsMinimumServerVersion(bucket, 5, 5)
+		if errServerVersion != nil {
+			return nil, errServerVersion
+		}
+
+		if !gsiSupported {
+			return nil, errors.New("Couchbase Server version must be 5.5 or higher for Sync Gateway to use GSI.  Upgrade the server, or set 'use_views':true in Sync Gateway's database config.")
```
Comment (tleyden) [inline rest/server_context.go:486]: I like how this lets the user know exactly how to fix the issue.

File: base/util_test.go [hunk 2]
```diff
@@ -291,3 +291,24 @@ func TestCouchbaseURIToHttpURL(t *testing.T) {
 	assert.True(t, err != nil)
 
 }
+
+// IsMinimumVersion takes (major, minor, minimumMajor, minimumMinor)
+func TestIsMinimumVersion(t *testing.T) {
+
+	// Expected true
+	assertTrue(t, isMinimumVersion(1, 0, 0, 0), "Invalid minimum version check - expected false")
```
Comment (tleyden) [inline base/util_test.go]: Shouldn't this say "expected true"?

Comment (adamcfraser) [reply base/util_test.go]: Good catch - thanks.

**Pull request merged**